### PR TITLE
Replacing rest_framework_jwt with django-rest-framework-simplejwt

### PR DESCRIPTION
**Pull request by alichass**
This is based on the fork of django-rest-auth by @ankurpandeyvns. 
This seems worth doing because rest_framework_jwt is no longer being maintained and it looks like drf-simple-jwt might end up being maintained by Jazzband https://github.com/SimpleJWT/django-rest-framework-simplejwt/issues/207

it seems to pass the current range of tests already implemented by dj-rest-auth as pretty much a drop-in for rest_framework_jwt

### CODE REVIEW

**Comment by ankurpandeyvns** [comment]
Glad to be a contributor! :)

**Comment by mohmyo** [comment]
I would love to see django-rest-framework-simplejwt being used

**Comment by filippo-20tab** [comment]
Same here! Thanks.

> I would love to see django-rest-framework-simplejwt being used

**Comment by robypomoni** [comment]
Same here! Thanks.

File: dj_rest_auth/views.py [hunk 1]
```diff
@@ -82,15 +83,6 @@ def get_response(self):
                                           context={'request': self.request})
 
         response = Response(serializer.data, status=status.HTTP_200_OK)
-        if getattr(settings, 'REST_USE_JWT', False):
```
[the remark this answers is not in the document]
Comment (alichass) [reply dj_rest_auth/views.py]: it seems like http-only cookies aren't yet implemented in simplejwt yet
https://github.com/SimpleJWT/django-rest-framework-simplejwt/issues/71

[the remark this answers is not in the document]
Comment (alichass) [reply dj_rest_auth/views.py]:
```python
if getattr(settings, 'REST_USE_JWT', False):
            from rest_framework_simplejwt.settings import api_settings as jwt_settings
            #if jwt_settings.JWT_AUTH_COOKIE        #this needs to be added to simplejwt or replaced with settings.JWT_AUTH_COOKIE
            from datetime import datetime
            expiration = (datetime.utcnow() + jwt_settings.ACCESS_TOKEN_LIFETIME)
            response.set_cookie(jwt_settings.JWT_AUTH_COOKIE,   #add to simpleJWT or replace with  settings.JWT_AUTH_COOKIE
                                self.access_token,
                                expires=expiration,
                                httponly=True)
```

As it stands, simplejwt doesnt seem to have support for httpOnly cookie storage
https://github.com/SimpleJWT/django-rest-framework-simplejwt/issues/71

I have a solution that i believe will work but that i havent tested thoroughly: the code below, and adding JWT_AUTH_COOKIE to either simplejwt's settings or this app's settings

```python
class JWTCookieAuthentication(JWTAuthentication):
    """
    An authentication plugin that hopefully authenticates requests through a JSON web
    token provided in a request cookie (and through the header as normal, with a preference to the header).
    """
    def authenticate(self, request):
        header = self.get_header(request)
        if header is None:
            if api_settings.JWT_AUTH_COOKIE: # or settings.JWT_AUTH_COOKIE
                raw_token = request.COOKIES.get(api_settings.JWT_AUTH_COOKIE)  # or settings.jwt_auth_cookie
            else:
                return None
        else:
            raw_token = self.get_raw_token(header)
            
        if raw_token is None:
            return None

        validated_token = self.get_validated_token(raw_token)

        return self.get_user(validated_token), validated_token
```
alternatively
```python 
class JWTCookieAuthentication(JWTAuthentication):
    """
    An authentication plugin that hopefully authenticates requests through a JSON web
    token provided in a request cookie 
    """
    def authenticate(self, request):
        raw_token = request.COOKIES.get(api_settings.JWT_AUTH_COOKIE)  # or settings.jwt_auth_cookie

        if raw_token is None:
            return None

        validated_token = self.get_validated_token(raw_token)

        return self.get_user(validated_token), validated_token
```

File: dj_rest_auth/views.py [hunk 2]
```diff
@@ -133,10 +125,6 @@ def logout(self, request):
 
         response = Response({"detail": _("Successfully logged out.")},
                             status=status.HTTP_200_OK)
-        if getattr(settings, 'REST_USE_JWT', False):
```
[the remark this answers is not in the document]
Comment (alichass) [reply dj_rest_auth/views.py]:
```python
if getattr(settings, 'REST_USE_JWT', False):
    from rest_framework_simplejwt.settings import api_settings as jwt_settings
    #if jwt_settings.JWT_AUTH_COOKIE        #this needs to be added to simplejwt
    response.delete_cookie('somestring') #replace with jwt_settings.JWT_AUTH_COOKIE
```

File: dj_rest_auth/views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alichass) [reply dj_rest_auth/views.py]: I think the new commit i made should allow it to work without any alterations to simplejwt, but I haven't really had the time to test it properly

**Comment by iMerica** [comment]
I've pushed a complete implementation for cookies with some additional unit tests.

**Pull request merged**